SwiftReflector/NewClassCompiler.cs

```
@@ -4477,6 +4478,86 @@ static int MachinePointerSize (MachO.Architectures arch)
+					var ex = ErrorHelper.CreateError (ReflectorError.kCompilerReferenceBase + 87, $"Unable to find matching constructor declaration for {tlf.MangledName} in enum {enumDecl.ToFullyQualifiedName ()}, skipping.");
+					errors.Add (ex);
+					continue;
+				}
+				var wrapperFunc = FindEquivalentFunctionDeclarationForWrapperFunction (wrapperFunction, TypeMapper, wrapper);
+				if (wrapperFunc == null) {
+					var ex = ErrorHelper.CreateError (ReflectorError.kCompilerReferenceBase + 88, $"Unable to find FunctionDeclaration for wrapper function {wrapperFunction.MangledName} in enum {enumDecl.ToFullyQualifiedName ()}, skipping.");
+					errors.Add (ex);
+					continue;
+				}
+				try {
+					var piConstructor = TLFCompiler.CompileMethod (wrapperFunc, use,
+						libraryPath, wrapperFunction.MangledName, pinvokeConsName, true, true, false);
+
+					var publicCons = TLFCompiler.CompileMethod (recastCtor, use, libraryPath, wrapperFunction.MangledName,
+						consName, isPinvoke: false, isFinal: true, isStatic: true);
+					var piCCTorName = PICCTorName (classType.ClassName);
+
+					var swiftCons = tlf.Signature as SwiftConstructorType;
+
+					var localIdents = new List<String> {
+						publicCons.Name.Name,
+						piConstructor.Name.Name,
+						piCCTorName,
+						kThisName
+					};
+
+					var engine = new MarshalEngine (use, localIdents, TypeMapper, wrapper.Module.SwiftCompilerVersion);
+
+					publicCons.Body.AddRange (engine.MarshalFunctionCall (wrapperFunc, false, pinvokeConsRef,
+						publicCons.Parameters, enumDecl, funcDecl.ReturnTypeSpec, csReturnType, swiftInstanceType: null,
+						instanceType: null, includeCastToReturnType: true, wrapper, includeIntermediateCastToLong: true));
+
+					en.Methods.Add (publicCons);
+					picl.Methods.Add (piConstructor);
+				} catch (Exception err) {
+					var ex = ErrorHelper.CreateError (ReflectorError.kCompilerReferenceBase + 88, err, $"Exception thrown marshaling to swift in enum constructor {recastCtor}");
```

You're swallowing this exception (nothing happens with the return value from CreateError), is that intentional?

---

👍